src/device.rs

```diff
@@ -617,6 +621,10 @@ impl Device {
     /// # let file = File::open("/dev/input/event0").unwrap();
     /// let device = uninit_device.set_file(file);
     /// ```
+    /// The caller is responsible for opening the file and setting
+    /// the `O_NONBLOCK` flag and handling permissions.
+    /// If the file is opened without O_NONBLOCK flag then next_event
+    /// should be called with ReadFlag::BLOCKING.
```

Comment:
I would add something like:

> If the file is not opened with `O_NONBLOCK` and `next_event()` is not called with `ReadFlag::BLOCKING`, then a call to `next_event()` may block trying to buffer events that have not yet happened.

so it is clear why those flags should be used

Reply:
Yes it would be good to capture this information